deps/rabbit/src/rabbit_quorum_queue.erl

```diff
@@ -713,59 +713,56 @@ restart_server({_, _} = Ref) ->
 -spec delete(amqqueue:amqqueue(),
              boolean(), boolean(),
              rabbit_types:username()) ->
-    {ok, QLen :: non_neg_integer()} |
-    {protocol_error, Type :: atom(), Reason :: string(), Args :: term()}.
-delete(Q, true, _IfEmpty, _ActingUser) when ?amqqueue_is_quorum(Q) ->
-    {protocol_error, not_implemented,
-     "cannot delete ~ts. queue.delete operations with if-unused flag set are not supported by quorum queues",
-     [rabbit_misc:rs(amqqueue:get_name(Q))]};
-delete(Q, _IfUnused, true, _ActingUser) when ?amqqueue_is_quorum(Q) ->
-    {protocol_error, not_implemented,
-     "cannot delete ~ts. queue.delete operations with if-empty flag set are not supported by quorum queues",
-     [rabbit_misc:rs(amqqueue:get_name(Q))]};
-delete(Q, _IfUnused, _IfEmpty, ActingUser) when ?amqqueue_is_quorum(Q) ->
+    {ok, QLen :: non_neg_integer()}.
+delete(Q, IfUnused, IfEmpty, ActingUser) when ?amqqueue_is_quorum(Q) ->
     {Name, _} = amqqueue:get_pid(Q),
     QName = amqqueue:get_name(Q),
     QNodes = get_nodes(Q),
-    %% TODO Quorum queue needs to support consumer tracking for IfUnused
-    Timeout = ?DELETE_TIMEOUT,
-    {ok, ReadyMsgs, _} = stat(Q),
-    Servers = [{Name, Node} || Node <- QNodes],
-    case ra:delete_cluster(Servers, Timeout) of
-        {ok, {_, LeaderNode} = Leader} ->
-            MRef = erlang:monitor(process, Leader),
-            receive
-                {'DOWN', MRef, process, _, _} ->
+
+    {ok, ReadyMsgs, Consumers} = stat(Q),
+    IsEmpty = ReadyMsgs == 0,
+    IsUnused = Consumers == 0,
+    if IfEmpty  and not(IsEmpty)  -> {error, not_empty};
```

Comment:
There is no need to line up bits of code like that. Please only use a single white space between things.
If you want the start of clauses to line up in columns use newlines.